File: 1046-MaxConsecutiveOnesIii/1046-MaxConsecutiveOnesIii.java
// Last updated: 07/04/2026, 11:37:14
class Solution {
    public int longestOnes(int[] nums, int k) {
        int n = nums.length;
        int ans =0 ;
        int skip = 0;
        int right = 0 ;
        int left = 0;
        for ( ; right<n;right++){
            if ( (nums[right] == 1) || (nums[right] == 0 && skip<=k)){
                ans = Math.max(ans,right-left);
                if ( nums[right] ==0) skip++;
            }
            while( skip>k){
                if ( nums[left] == 0) skip--;
                left++;
            }
        }
        ans = Math.max(ans,right-left);
        return ans;

    }
}

File: 1046-MaxConsecutiveOnesIii/1046-MaxConsecutiveOnesIii.cpp
// Last updated: 07/04/2026, 11:37:19
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int ans = 0;
        int n = nums.size();
        int left = 0 ; 
        int skip =0;
        
        int right = 0;

        for (;right <n;right++){

            if ( nums[right] == 1){
                ans = max(ans,right-left);
            }
            else if ( nums [right] == 0 && skip <=k){
                ans = max(ans,right-left);
                skip++;
            }
            while(skip >k){
                if ( nums[left] == 0) skip--;
                left++;

            }

        }
        ans = max(ans,right-left);
        return ans;

    }
};